beacon-chain/core/validators/validator.go

```diff
@@ -403,15 +383,56 @@ func ProcessPenaltiesAndExits(ctx context.Context, state *pb.BeaconState) *pb.Be
 			totalAtEnd := state.LatestSlashedBalances[slashedEpoch]
 			totalPenalties := totalAtStart - totalAtEnd
 
-			penaltyMultiplier := totalPenalties * 3
-			if totalBalance < penaltyMultiplier {
-				penaltyMultiplier = totalBalance
+			var penalty uint64
+			index := uint64(idx)
+            effectiveBalance := helpers.EffectiveBalance(state, index)
+            minPenaltyQuotient := params.BeaconConfig().MinPenaltyQuotient
+            if totalPenalties*3 < totalBalance {
+				if effectiveBalance*totalPenalties*3/totalBalance > effectiveBalance/minPenaltyQuotient {
```

Comment:
This formatting here looks a little weird, think you can align it with the rest? 🙂

Reply:
+1